### DISCUSSION TITLE
Error code

### DISCUSSION THREAD

**Discussion by jlps5702** [aclap-dev/video-downloadhelper#519] [Help and Q&A]
### Your question

{
  "version": "9.0.1.13a1",
  "target": "mozilla",
  "channel": "beta",
  "page_url": "https://missav.com/ebwh-102",
  "error": {
    "error": "coapp_failure",
    "details": "SideDownload error: : [in#0 @ 000002bc34e7ac00] Error opening input: Server returned 403 Forbidden (access denied)\r\nError opening input file https://surrit.com/34129b2d-d553-4c6e-9938-0449276d3657/720p/video.m3u8.\r\nError opening input files: Server returned 403 Forbidden (access denied)\r\n",
    "id": "coapp_failure_78b20adc-67eb-4eb5-b429-f81472503a60",
    "downloadable_id": "downloadable_7177404397799402"
  }
}

### Addon & CoApp version and details

{
  "version": "9.0.1.13a1",
  "target": "mozilla",
  "channel": "beta",
  "page_url": "https://missav.com/ebwh-102",
  "error": {
    "error": "coapp_failure",
    "details": "SideDownload error: : [in#0 @ 000002bc34e7ac00] Error opening input: Server returned 403 Forbidden (access denied)\r\nError opening input file https://surrit.com/34129b2d-d553-4c6e-9938-0449276d3657/720p/video.m3u8.\r\nError opening input files: Server returned 403 Forbidden (access denied)\r\n",
    "id": "coapp_failure_78b20adc-67eb-4eb5-b429-f81472503a60",
    "downloadable_id": "downloadable_7177404397799402"
  }
}

**Comment by GrampaWildWilly** [comment]
I am running Windows 11 Pro 64-bit version 23H2 build 22631.3593, Firefox 126.0 64-bit, licensed VDH 9.0.1.13a1 beta, CoApp 2.0.19.  I do not work for VDH.  I'm just another user, like you.

I visited the web page mentioned in your problem report as the page_url.

WARNING

This is a Japanese porn site.

I opened the VDH menu & I was stunned by the amount of updating going on in it.  There were new entries being added to the menu at a rate of at least 1 per second.  The menu was many scrolls long when I first opened it & it grew at a breakneck speed as I was looking at it.  However, you can deal with it but you have to be quick about using the menu.  You need a combination of keyboard manipulation & mouse manipulation.  The keyboard manipulation is this: Open the menu, Tab, End.  That puts the menu at the bottom.  But it will be the bottom for only a moment since that web page bombards the client browser with endless junk.  I suspect it's advertising.  It's quite heinous.  Anyway, press & hold down the End key so the menu stays scrolled to the bottom.  Way down there, if you can get your eyes to focus on the constantly flickering menu, you will see 2 entries showing durations of 2:16:17, which is the duration of this video.  One of those entries is marked HLS, the other is marked Raw HLS.  You want the HLS entry, not the Raw HLS entry.  This is, by the way, a general rule of thumb.  VDH often shows you both Raw HLS & simply HLS.  You always want to prefer the HLS entry.  In any case, while you are holding down the End key, you can manage to position your mouse cursor over the Download button in that HLS entry of the VDH menu.  Then you have to be really quick because you can't be holding down the End key when you click the mouse button.  So release End & click the mouse on the Download button.  And be quick about it.  This will get VDH to launch a download of the video.  That's how I downloaded this.  It took a bit under 2 minutes to download.  Resource Monitor was showing me a download speed of about 12 million bytes per second, which is very much above average.  You won't be able to monitor this in the VDH menu unless you do the scroll to bottom/End thing I described above.  It's easier to just watch the Resource Monitor.  At the end of this, I had a 1.41G MP4 of resolution 1280x720 that played just fine in VLC.  I didn't sit & watch this, just sampled it at intervals, & it seemed to be all there, audio & video, right to the end.  I would normally show some screenshots of all this (heavily censored, of course), but the incessant updates on the VDH menu made that impossible.

At the end of all this, your VDH menu will be full to overflowing.  I recommend you close the porn web page.  This will get the incessant updating to stop.  I would also recommend either clicking the Reload addon button in the VDH settings or momentarily disabling/re-enabling VDH in the browser settings.  This will clear all the leftover junk this web page will have been spewing at you.

@paulrouget Paul, you might want to look at this web page to observe the truly awful behavior of the VDH menu on this web page.  It might give you some ideas about how to deal with such pathological web pages.

**Comment by paulrouget** [comment]
@GrampaWildWilly I guess you enabled the "show low quality media" option. That's why this option is unchecked by default.

**Reply by GrampaWildWilly** [reply]
Oh.  OK.  I'll have to remember that.  Thanks.

**Reply by GrampaWildWilly** [reply]
Yes.  Much better.  I went back to the Japanese porn page & the VDH menu looked like this:

![image](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/18f709b5-38d5-478a-a167-95c7ba6cb988)

The entry with the right duration is the one I downloaded successfully earlier.

Except now, I'm getting the same error as the original poster.  So I switched the low quality media on & off, reloaded VDH, reloaded the web page, over & over.  The web page reloads were Ctrl+F5, bypassing my disk cache on my system.  After some large number of retries, with & without the low quality setting, it FINALLY downloaded again.  So it seems this web site is temperamental.  You have to keep reloading the web page & retrying in VDH.  I don't know how many times I reloaded the page, maybe 15 or 20, before it finally worked again.  And I was getting even better download speed than earlier, more like 16 million bytes per second.  I must have been lucky earlier because it worked on the first try that time.  So the moral of the story is that this site is tempermental, idiosyncratic, pathological.  You have to keep reloading the web page until VDH finally succeeds in downloading the content.  There have been a modest number of other reports in here of problems with this site.  I guess I've finally gotten to the bottom of it.  The solution is to just keep reloading the page until it works.  VDH will work.  You have to be persistent.  Maybe I was lucky that it took me only 15 or 20 retries before it worked.  You might get away with fewer.  You might need to do more.  I think the problem is with the web site, not VDH.  But you might want to investigate it further, Paul.

**Reply by GrampaWildWilly** [reply]
By the way, Paul, have you noticed that some of the URLs in the error reports are getting this appended?

.\r\nError

You can't just click on the URL.  You have to edit the URL before you can use it.  Rather tedious.

**Reply by GrampaWildWilly** [reply]
That's return, new line, isn't it?  It's not coded right.  It's not working.

**Reply by GrampaWildWilly** [reply]
Here's what it looks like on Windows:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/fbf3773c-7e16-4214-b2fd-3ba6800bb88d)

I'm guessing you aren't seeing this because you're developing on Linux.  Whatever is working for you on Linux is not working on Windows.

**Reply by paulrouget** [reply]
It's normal. These are carriage returns from a TTY io.

**Reply by GrampaWildWilly** [reply]
But they make the messages ugly & they mess up the URLs so you can't just click on them.  I can't accept that it's normal.  There are supposed to be line breaks there.  It's supposed to look like this:

"error": "coapp_failure",
"details": "SideDownload error: : [in#0 @ 000002bc34e7ac00] Error opening input: Server returned 403 Forbidden (access denied)
Error opening input file https://surrit.com/34129b2d-d553-4c6e-9938-0449276d3657/720p/video.m3u8
Error opening input files: Server returned 403 Forbidden (access denied)

Surely it's possible to program the error handler to make the error messages look like that.  TTY io is programmer-speak that is neither interesting to nor understood by ordinary end users.  You need to make the error messages come out properly.  The \r\n should never be visible.  There should just be line breaks there.

**Comment by jlps5702** [comment]
FIX?

    在 2024年5月21日 星期二 下午07:44:04 [GMT+8]， ***@***.***> 寫道：  
 
 


But they make the messages ugly & they mess up the URLs so you can't just click on them. I can't accept that it's normal. There are supposed to be line breaks there. It's supposed to look like this:

"error": "coapp_failure",
"details": "SideDownload error: : [in#0 @ 000002bc34e7ac00] Error opening input: Server returned 403 Forbidden (access denied)
Error opening input file https://surrit.com/34129b2d-d553-4c6e-9938-0449276d3657/720p/video.m3u8
Error opening input files: Server returned 403 Forbidden (access denied)

Surely it's possible to program the error handler to make the error messages look like that. TTY io is programmer-speak that is neither interesting to nor understood by ordinary end users. You need to make the error messages come out properly. The \r\n should never be visible. There should just be line breaks there.

**Reply by GrampaWildWilly** [reply]
If you're asking whether there's a fix for downloading from missav, I suspect there isn't a fix VDH can apply.  The problem is in the web site.

If you're asking for a fix to the ugly error messages, I welcome more votes from fellow users for Paul to correct the format of the error messages.  To me, TTY io is a pretty lame excuse.

**Comment by charlestp13** [comment]
I have issues with this website as well. Everything was easy on the first day of downloading. Overnight, something happened, and suddenly all full videos returned error 403. I can still view the videos fine, freely. Is there a workaround or have they put a permanent block on my Mac address etc... that renders downloading from them impossible?
